### PR TITLE
MdBook hosted on Github Pages via Github Actions

### PR DESCRIPTION
**Pull request by FongYoong**
1.  Made a minor adjustment to README.md as explained in my reply to [this](https://github.com/Dhghomon/easy_rust/issues/96#issuecomment-749523972).

2. `createGithubPagesFromReadme.sh` is a fork of `createBookFromReadme.sh`. This new Bash script does not manually build the MdBook. It only reuses the original script's code which splits README.md into multiple chapters. One downside to the old script is that the chapters are not sorted in order. Hence, `ls -tr | grep Chapter_ | sort -V` in line 38 ensures that the order is correct.

3. Makefile now contains `github_pages` which executes `createGithubPagesFromReadme.sh`. Usage: `make github_pages`

4. Added `github-pages.yml` to `.github/workflows` which configures Github Action to deploy a functional MdBook to Github Pages. If you're not acquainted with [Github Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions), it's basically a continuous integration (CI) system similar to Travis, CircleCI etc. Essentially. the `.yml` file instructs Github to install homebrew and coreutils which are dependencies for the splitting part, modify the scripts' permissions, run `make github_pages`, installs MdBook with [this](https://github.com/peaceiris/actions-mdbook), runs `mdbook build`, and finally deploys the docs to Github Pages.
Note that the docs are placed in a new branch, `gh-pages`, as shown below. You can ignore this branch and leave it alone.

image

***
5. Every time you push a change to the master branch, Github will automatically rebuild the docs which is pretty neat I think. Lastly, I added a badge in README.md so that you can easily see whether the build passes or vice versa.

6. In my fork, the built docs can be found at [https://fongyoong.github.io/easy_rust/](https://fongyoong.github.io/easy_rust/) which follows the format of [https://<your_username>.github.io/<repo_name>]().
If you can't find your own address, check the Settings of your repo to find the address. It should look like below:

image

***

7. For your info, Github imposes a limit on Actions. More info can be found [here](https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions).
However, I'm pretty sure it won't be an issue for you because the build times are usually around 1.5 minutes as shown below, and the book is pretty much complete for now.

image

***
8. This is actually my first pull request and there may be mistakes lying around. Thanks for drafting the book, it really eased my steps towards Rust. Cheers! 😄

### CODE REVIEW

File: createGithubPagesFromReadme.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    done
+}
+
+# Creates the summary from the generated chapters
+function createSummary(){
+    cd ./src
+    touch SUMMARY.md
+    echo '# Summary' > SUMMARY.md
+    echo "" >> SUMMARY.md
+    for f in $(ls -tr | grep Chapter_ | sort -V); do
```
Comment (FongYoong) [inline createGithubPagesFromReadme.sh:38]: Sorts the contents of the generated Summary.md

File: README.md [hunk 2]
```diff
@@ -1,4 +1,9 @@
-Update 28 November 2020: [Now also available in simplified Chinese](https://github.com/kumakichi/easy_rust_chs) thanks to [kumakichi](https://github.com/kumakichi)!
+## Update
```
Comment (FongYoong) [inline README.md:1]: Circumvents the bug stated in [this issue](https://github.com/Dhghomon/easy_rust/issues/96).

**Comment by Dhghomon** [comment]
Hi! Thanks for the PR. This is all outside of my area of expertise and I see it's your first PR too, but then again it's to fix an issue. Do you want anyone to take a second look first or should I just merge it now?

**Comment by FongYoong** [comment]
There shouldn't be a problem merging, given that there's no conflicts. It's your call.

**Comment by Dhghomon** [comment]
Sounds good! Here goes.

**Comment by FongYoong** [comment]
Thanks for the merge!

By the way, can you go to your repo's settings and activate GitHub Pages?

1. Go to **Settings**:

image

2. Scroll all the way down to **GitHub Pages** and change the settings as shown below:

image

3. Reply to me the **link** of the published website.

Sorry for the extra procedures. I need to make sure the online documentation is built from your repo, not mine. When you make further changes to your README.md, the online docs will automatically be updated as well.

**Comment by Dhghomon** [comment]
Hey,

Looks like it worked: https://dhghomon.github.io/easy_rust/

**Pull request merged**